src/main.rs

```diff
@@ -0,0 +1,106 @@
+use std::io::{self, Write};
+use regex::{self, Regex};
+
+enum MetaCommand{
+    Exit
+}
+
+impl MetaCommand{
+    fn is_meta_command(input_str: &str) -> bool {
+        input_str.starts_with('.')
+    }
+
+    fn categorize(input_str: &str) -> Option<Self>{
+        match input_str {
+            ".exit" => Some(MetaCommand::Exit),
+            _       => None
+        }
+    }
+}
```

Comment:
I think the idiomatic alternative to `categorize` is using the `FromStr` trait. Once you implement that you can use the `parse` method on any string to get a `MetaCommand` from it. Here's an example of what that would look like:

```suggestion
impl MetaCommand {
    fn is_meta_command(input_str: &str) -> bool {
        input_str.starts_with('.')
    }
}

enum MetaCommandParseError {
    UnrecognizedCommand,
}

impl FromStr for MetaCommand {
    type Err = MetaCommandParseError;

    fn from_str(input_str: &str) -> Result<Self, Self::Err> {
        match input_str {
            ".exit" => Ok(MetaCommand::Exit),
            _ => Err(MetaCommandParseError::UnrecognizedCommand),
        }
    }
}
```

I'm not sure about creating an enum for this specific error. Maybe you could just extend or repurpose your `ParseError`, idk :)